utils/useful.py

```diff
@@ -117,6 +140,8 @@ def actually_calls(param: tuple, callback: Callable, /) -> List[Any]:
         return_result = callback(array_string, content_buffer, size)
         return decode_result(return_result)
 
+    return []
```

Comment:
why

Reply:
because ` -> List[Any]`. I either do this or do `-> Optional[List[Any]]` and `return`

Reply:
Make it Optional[List[Any]] and dont return a list